Allow to disable graphs and set number of days of historical data for services

CSharpRU opened the pull request:
Hello there,

I've added ability to switch off/on graphs for services and ability to set number of days of historical data to retrieve from backend for services:

<img width="1081" alt="image" src="https://user-images.githubusercontent.com/640600/196636351-25573e51-bfab-4517-973a-4297a0f39a47.png">

<img width="993" alt="image" src="https://user-images.githubusercontent.com/640600/196636277-3c840c49-da5b-46cc-8f91-7b1b58b24f21.png">

<img width="724" alt="image" src="https://user-images.githubusercontent.com/640600/196637749-152e6a7d-1d93-4913-8e74-dc4820481898.png">

Tested locally, all new features work as intended. Please advice in code, logic or missed things that I need to fix/add :)

Fixes? https://github.com/statping-ng/statping-ng/issues/96

CSharpRU committed:
allow to disable graphs and set number of days of historical data for services

jemand771 commented:
haven't looked at all of the code yet, but as some quick feedback:
* your PR has big whitespace changes in some files (diff viewer shows big diff but nothing changed) - maybe you can fix those so it's easier to see what you did
* I like the idea of globally disabling all graphs, but I could also see someone wanting to hide them on a per-service basis. how do you feel about that idea?
* nitpick: I'd probably call the option "show graphs" or "enable graphs" and have it on by default instead of "disables graphs" and having it off by default :b

CSharpRU commented:
> haven't looked at all of the code yet, but as some quick feedback:
> 
> * your PR has big whitespace changes in some files (diff viewer shows big diff but nothing changed) - maybe you can fix those so it's easier to see what you did
> * I like the idea of globally disabling all graphs, but I could also see someone wanting to hide them on a per-service basis. how do you feel about that idea?
> * nitpick: I'd probably call the option "show graphs" or "enable graphs" and have it on by default instead of "disables graphs" and having it off by default :b

Thanks, I’ll look into that and change the code.

CSharpRU committed:
rename disable_graphs to show_graphs and add ability to on/off graph per service

CSharpRU commented:
Renamed disable graphs to show graphs and set enabled by default:

<img width="682" alt="image" src="https://user-images.githubusercontent.com/640600/196881813-6ffe5353-694a-4d32-9983-21060dbff5f8.png">

Added ability to disable graph per service:

<img width="969" alt="image" src="https://user-images.githubusercontent.com/640600/196892973-88507236-dd67-44eb-accc-7413feaba460.png">

> * your PR has big whitespace changes in some files (diff viewer shows big diff but nothing changed) - maybe you can fix those so it's easier to see what you did

I don't see any issues in VS Code or GitHub, can you give me an example or screenshot? :)

CSharpRU committed:
Merge branch 'dev' into dev

jemand771 commented:
nevermind on the whitespace, those are all good. you indented a lot of things because `NumberOfDaysForService` is longer than all of the other settings, but that's fine
(I initially thought this might be a line ending or tabs/spaces thing)

CSharpRU commented:
I think that's it then or should I add something else?

CumpsD commented:
This is a nice feature!

bretdavi89 commented:
Not to be that guy, but wondering what the status of reviewing/merging this PR is? This would be a very nice feature to have.

bretdavi89 commented:
Pulled this into my fork and noticed that the status bar doesn't appropriately resize for shorter day ranges. E.g. had this set to 14 days:
![image](https://user-images.githubusercontent.com/72528410/217039775-96537f07-0c8e-4893-8bfe-1b110d5eee3e.png)

So appears that scaling of those elements needs to occur to fill the row

CSharpRU commented:
> Pulled this into my fork and noticed that the status bar doesn't appropriately resize for shorter day ranges. E.g. had this set to 14 days: ![image](https://user-images.githubusercontent.com/72528410/217039775-96537f07-0c8e-4893-8bfe-1b110d5eee3e.png)
> 
> So appears that scaling of those elements needs to occur to fill the row

I haven't touched frontend for 8 years, so if anyone can help that'd be great :D

Appel-flappen commented:
This would be a great feature, also would fix the issue on mobile where the little status pills become tiny and antialias weird, giving different sizes.

(oops, didn't mean to review and accept)

Grunticus03 commented:
Sooo....can this be merged?
